src/main/java/database/Database.java

```diff
@@ -54,26 +74,52 @@ public synchronized void remove(AppointmentId id, AppointmentType type) {
     }
   }
 
+  public List<Appointment> findById(PatientId pid) {
+    List<Appointment> appointments = null;
+    hashMap.values().forEach(mmp-> mmp.values().forEach(mmp2-> {
+              ArrayList<PatientId> ids= mmp2.getPatientIds();
+              for(PatientId idd:ids){
+                if (idd == pid) {
+                  AppointmentId appointId = mmp2.getAppointmentId();
+                  var appointment = mmp.get(appointId);
+                  appointments.add(appointment);
+
+                }
+              }
+            })
+    );
+    if (appointments != null) {
+      return appointments;
+    } else {
+      return Collections.emptyList();
+    }
```

Comment:
Just return `appointments`. The list will be empty if no appointment is found. 
Left it with its caller.